Index: linyaps/libs/linglong/src/linglong/repo/ostree_repo.cpp
===================================================================
--- linyaps.orig/libs/linglong/src/linglong/repo/ostree_repo.cpp
+++ linyaps/libs/linglong/src/linglong/repo/ostree_repo.cpp
@@ -1270,6 +1270,28 @@
     return LINGLONG_OK;
 }
 
+// 执行apt-config 获取 apt token配置
+utils::error::Result<std::string> getAptToken() noexcept
+{
+    LINGLONG_TRACE("get apt token");
+    auto aptToken =
+      linglong::utils::command::Cmd("apt-config").exec({ "shell", "Token", "Acquire::SmartMirrors::Token" });
+
+    if (!aptToken) {
+        return LINGLONG_ERR(aptToken);
+    }
+    auto token = (*aptToken).toStdString();
+    if (token.empty()) {
+        return LINGLONG_ERR("not found token");
+    }
+    auto start = token.find("'");
+    auto end = token.rfind("'");
+    if (start != std::string::npos && end != std::string::npos && end > start) {
+        return token.substr(start + 1, end - start - 1);
+    }
+    return LINGLONG_ERR("token format error");
+}
+
 // 初始化一个GVariantBuilder
 GVariantBuilder OSTreeRepo::initOStreePullOptions(const std::string &ref) noexcept
 {
@@ -1290,6 +1312,18 @@
                           "{s@v}",
                           "refs",
                           g_variant_new_variant(g_variant_new_strv(refs.data(), -1)));
+    auto token = getAptToken();
+    if (!token) {
+        LogW("get apt token error: {}", token.error().message());
+    } else {
+        GVariantBuilder hdr_builder;
+        g_variant_builder_init(&hdr_builder, G_VARIANT_TYPE("a(ss)"));
+        g_variant_builder_add(&hdr_builder, "(ss)", "X-Repo-Token", token->c_str());
+        g_variant_builder_add(&builder,
+                              "{s@v}",
+                              "http-headers",
+                              g_variant_new_variant(g_variant_builder_end(&hdr_builder)));
+    }
     return builder;
 }
 
